app/src/processing/app/Base.java

```diff
@@ -1360,6 +1367,26 @@ public void onBoardOrPortChange() {
       }
     }
 
+    // Update recent boards list in preferences
+    List<String> newRecentBoardIds = new ArrayList<String>();
+    String currentBoard = PreferencesData.get("board");  
+    for (String recentBoard : PreferencesData.getCollection("recent.boards")){
+      if (!recentBoard.equals(currentBoard)) {
+        newRecentBoardIds.add(recentBoard);
+      }
+    }
```

Comment:
I'll look into that. I'm actually not that familiar with Java, this was my first crack at it. I suspect there are a lot of optimizations to do with collections since I mostly treated them like arrays.

Reply:
I can't figure out the optimization for this without breaking it.

Reply:
I implemented this in my branch: https://github.com/matthijskooijman/Arduino/commit/818c3fe3a497ea75737ff83c0631b943fa0f8413